components/browser/engine-servo/src/main/java/mozilla/components/browser/engine/servo/ServoEngineSession.kt

```diff
@@ -0,0 +1,150 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.servo
+
+import android.graphics.Bitmap
+import android.net.Uri
+import mozilla.components.concept.engine.DefaultSettings
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.concept.engine.Settings
+import org.mozilla.servoview.Servo
+import org.mozilla.servoview.ServoView
+import java.lang.ref.WeakReference
+
+/**
+ * Servo-based EngineSession implementation.
+ */
+@Suppress("TooManyFunctions")
+class ServoEngineSession(
+    private val defaultSettings: Settings? = null
+) : EngineSession() {
+    override val settings: Settings
+        get() = defaultSettings ?: DefaultSettings()
+
+    private var urlToLoad: String? = null
+    private var viewReference: WeakReference<ServoEngineView>? = null
+    private val view: ServoView?
```

Comment:
Isn't that the same with the `view` property here?

Reply:
Yes, but since it's a private field, mocking that wouldn't be as easy, right? 

Maybe I'm just a bit more fond of our `SystemEngineSession#currentView` after trying to write a test for the exitFullScreen callback to cover all the branches with something like `weakRef?.get()?.anotherNullableObject?.invoke()`.

```kotlin
internal fun currentView(): WebView? {
        return view?.get()?.currentWebView
}
```

With that, it's easier to just mock that function and let those branches not add to the branches that we can't cover (because mocking WeakReference is hard).

This is also just opinion. 👌